atlasdb-client/src/main/java/com/palantir/atlasdb/logging/LoggingArgs.java

```diff
@@ -82,13 +83,28 @@ public static SafeAndUnsafeTableReferences tableRefs(Collection<TableReference>
                 .build();
     }
 
+    public static List<String> tableRefValues(Collection<TableReference> tables) {
+        return tables.stream().map(LoggingArgs::tableRefValue).collect(Collectors.toList());
+    }
+
     /**
      * Returns a safe or unsafe arg corresponding to the supplied table reference, with name "tableRef".
      */
     public static Arg<String> tableRef(TableReference tableReference) {
         return tableRef("tableRef", tableReference);
     }
 
+    /**
+     * If safe, returns the table name. If unsafe, returns {table}.
+     */
+    public static String tableRefValue(TableReference tableReference) {
+        if (logArbitrator.isTableReferenceSafe(tableReference)) {
+            return tableReference.toString();
+        } else {
+            return "{table}";
```

Comment:
We should avoid the `toString()` allocation overheads when not tracing, would propose we keep these as table refs and use a placeholder for unsafe ones, something along lines of:

```
    private static final TableReference PLACEHOLDER_TABLE_REFERENCE = TableReference.create(Namespace.DEFAULT_NAMESPACE, "{table}");

    public static TableReference tableRefValue(TableReference tableReference) {
        if (logArbitrator.isTableReferenceSafe(tableReference)) {
            return tableReference;
        } else {
            return PLACEHOLDER_TABLE_REFERENCE;
        }
    }
```